clang-tools-extra/include-cleaner/lib/WalkAST.cpp

```diff
@@ -203,7 +203,7 @@ class ASTWalker : public RecursiveASTVisitor<ASTWalker> {
   bool VisitUsingDecl(UsingDecl *UD) {
     for (const auto *Shadow : UD->shadows()) {
       auto *TD = Shadow->getTargetDecl();
-      auto IsUsed = TD->isUsed() || TD->isReferenced();
+      auto IsUsed = TD->isUsed() || TD->isReferenced() || !TD->getAsFunction();
```

Comment:
nit: the name `IsUsed` no longer reflects its purpose. Consider renaming it to something more suitable or inlining it.

Additionally, it would be helpful to add comments to document the intention behind this code.

Reply:
Well I was actually thinking that `usedness` is still the right concept here, spelling of the type name in the using declaration is enough to trigger that use for records.

I added comments also along those lines, LMK if it still doesn't resonate with you.

Reply:
The connection between usedness and `!TD->getAsFunction()` wasn’t immediately obvious to me. With the newly-added comment, it's much clearer now, everything looks good, thanks.